Rakefile

```diff
@@ -0,0 +1,24 @@
+require 'asciidoctor'
+require 'asciidoctor/doctest'
+require 'thread_safe'
+require 'tilt'
+
+DocTest::RakeTasks.new do |t|
+  t.output_examples :html, path: 'test/output/slim'
+  t.input_examples :asciidoc, path: [ *DocTest.examples_path, 'examples' ]
+  t.converter = DocTest::HTML::Converter
+  t.converter_opts = { template_dirs: 'templates/slim' }
+end
+
+desc 'Renders all the test slides into fully working examples that you can look in a browser'
+# rendered slides will be put in examples/ directory
+task :render do
+#    Asciidoctor.convert_file 'data-background-newstyle.adoc',
+#        :backend => 'revealjs',
+#        :base_dir => 'examples'
+##        :template_dir => 'templates/slim'
+	system "bundle exec asciidoctor-revealjs data-background-oldstyle.adoc", :chdir => 'examples'
+end
```

Comment:
@Mogztter: I didn't manage to make `convert_file` work. `base_dir` was not honored. Any ideas?

Reply:
I'm guessing that you may have to configure the  `safe` mode.

Reply:
The following seems to work:

```ruby
Asciidoctor.convert_file 'data-background-newstyle.adoc',
        :safe => 'safe',
        :backend => 'revealjs',
        :base_dir => 'examples'
```

Reply:
Still not working for me:
```
$ bundle exec rake render 
rake aborted!
Errno::ENOENT: No such file or directory @ rb_sysopen - data-background-newstyle.adoc
/home/olivier/src/asciidoc/asciidoctor-reveal.js/.bundle/gems/ruby/2.4.0/gems/asciidoctor-1.5.6.1/lib/asciidoctor.rb:1575:in `initialize'
/home/olivier/src/asciidoc/asciidoctor-reveal.js/.bundle/gems/ruby/2.4.0/gems/asciidoctor-1.5.6.1/lib/asciidoctor.rb:1575:in `open'
/home/olivier/src/asciidoc/asciidoctor-reveal.js/.bundle/gems/ruby/2.4.0/gems/asciidoctor-1.5.6.1/lib/asciidoctor.rb:1575:in `convert_file'
/home/olivier/src/asciidoc/asciidoctor-reveal.js/Rakefile:19:in `block in <top (required)>'
/usr/bin/bundle:22:in `load'
/usr/bin/bundle:22:in `<main>'
Tasks: TOP => render
(See full trace by running task with --trace)
```

Should I change `cwd` with Rake? Anyway, I want the output to be in `examples/` and gitignore `examples/reveal.js/`.

Reply:
Uh oh I see... I didn't run the right Rake task 😳 
What about:
```ruby
Asciidoctor.convert_file 'examples/data-background-newstyle.adoc',
        :safe => 'safe',
        :backend => 'revealjs',
        :base_dir => 'examples'
```

`to_dir` should be relative to `base_dir` so the document should be rendered in the `examples/` directory.

Reply:
If you want the document to be rendered in `examples/reveal.js/`, I think you need to define `:to_dir => 'reveal.js'`.